File: src/main/java/com/github/alexandrecarlton/idea/settings/starter/IdeaModule.java
package com.github.alexandrecarlton.idea.settings.starter;

import com.intellij.openapi.application.ex.ApplicationEx;
import com.intellij.openapi.application.ex.ApplicationManagerEx;

import dagger.Module;
import dagger.Provides;

/**
 * Binds isolated singleton instances of IntelliJ IDEA.
 */
@Module
public class IdeaModule {

  @Provides
  static ApplicationEx provideApplicationEx() {
    return ApplicationManagerEx.getApplicationEx();
  }

}

File: dagger/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/IdeaModule.kt
package com.github.alexandrecarlton.idea.settings.dialog

import dagger.Binds
import dagger.Module

@Module
interface IdeaModule {

    @Binds
    fun bindIdeaSettingsApplier(applier: IdeaSettingsApplier): SettingsApplier<IdeaSettings>

}
